src/Ocelot.Provider.Kubernetes/EndPointClientV1.cs

```diff
@@ -30,8 +33,28 @@ public async Task<EndpointsV1> GetAsync(string serviceName, string kubeNamespace
 
         var response = await Http.GetAsync(request, cancellationToken);
 
-        return response.IsSuccessStatusCode
-            ? await response.ReadContentAsAsync<EndpointsV1>()
-            : null;
+        if (!response.IsSuccessStatusCode)
+        {
+            if (_logger.IsEnabled(LogLevel.Debug))
```

Comment:
I am not certain, but if we use the native Ocelot logger, then all `Log*` methods have internal check to determine whether logging is enabled. Therefore, this `if` clause may be redundant.

Reply:
Yeah, ILogger does the same, but this check was more to decide if we want to check whether to try deserialising the error response;  it’s a pattern I tend to use when we perform some non-zero-cost operation to collect info purely for logging purposes. Some rationale from the runtime folks here:

https://github.com/dotnet/runtime/issues/45290#issue-752502603

Reply:
Having said that, as designed, this method swallows any error and returns null (logging being the only place errors are surfaced) so perhaps a warning is more appropriate? just didn’t want to fill up the log with noise… 🤷🏻

Reply:
I've moved that logic out of `EndPointClientV1`, to maintain separation of concerns; it now lives in the caller (`Kube`).